src/samplebuffer.h

```diff
@@ -0,0 +1,90 @@
+#ifndef SAMPLEBUFFER_H
+#define SAMPLEBUFFER_H
+
+#include "util/types.h" // CSAMPLE
+
+#include <algorithm> // std::swap
+
+// A sample buffer with properly aligned memory to enable SSE optimizations.
+//
+// No resize operation is provided intentionally for maximum efficiency!
+// If the size of an existing sample buffer needs to be altered after
+// construction this can simply be achieved by swapping the contents with
+// a temporary sample buffer that has been constructed with the appropriate
+// size:
+//
+//     SampleBuffer sampleBuffer(oldSize);
+//     ...
+//     SampleBuffer(newSize).swap(sampleBuffer);
+//     // Now the data in sampleBuffer is uninitialized and
+//     // sampleBuffer.size() == newSize
+//
+// No sample data is copied when swapping the contents of two samples buffers
+// for resizing!
+class SampleBuffer {
+    Q_DISABLE_COPY(SampleBuffer);
+
+  public:
+    typedef size_t size_type;
+    typedef CSAMPLE value_type;
```

Comment:
Well, it actually does or did hurt a little bit when using such a member typedef as a return type of a function. But this will be fixed with the new (trailing) return value syntax of C++11.